### PR TITLE
remove pytest-asyncio

### PR DESCRIPTION
**Pull request by graingert**
gen_test doesn't transport marks correctly and so must be the first
decorator applied to a test function

also gen_test is stricter than pytest-asyncio it checks for leaky comms and
threads, so we need to make sure resources are closed properly
- [x] Closes https://github.com/dask/distributed/issues/6050
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: distributed/comm/tests/test_ucx.py
Comment (graingert) [inline distributed/comm/tests/test_ucx.py:299]:
```suggestion
@gen_test(timeout=60)
```

File: distributed/deploy/tests/test_local.py
Comment (graingert) [inline distributed/deploy/tests/test_local.py:1061]: the `asynchronous=True` kwarg was missing so this spawned a new thread

**Comment by graingert** [comment]
the macos failures seem unrelated

File: distributed/deploy/tests/test_local.py
Comment (sjperkins) [inline distributed/deploy/tests/test_local.py:1074]: This is probably a stray print?

File: distributed/deploy/tests/test_local.py
Comment (sjperkins) [inline distributed/deploy/tests/test_local.py:1067]: I guess this is removed because the `async with cluster` in the modified version reliably blocks for this test now?

File: distributed/comm/tests/test_ucx.py
Comment (sjperkins) [inline distributed/comm/tests/test_ucx.py:299]: Why is the timeout necessary? I wonder if this test gets into blocked states which necessitate the timeout?

File: distributed/tests/test_batched.py
Comment (sjperkins) [inline distributed/tests/test_batched.py:65]: This seems like a good catch :)

File: distributed/tests/test_core.py
Comment (sjperkins) [inline distributed/tests/test_core.py:713]: +1 for resource cleanup

File: distributed/tests/test_preload.py
Comment (sjperkins) [inline distributed/tests/test_preload.py:191]: This seems to be a  more robust expression of the previous test, achieved by:

1. Avoiding creating the preload application in a separate process
2. Allowing a dynamically allocated port

File: distributed/tests/test_worker.py
Comment (sjperkins) [inline distributed/tests/test_worker.py:1361]: cleanup used to call `gc.collect()`. Is there a particular reason this fixture isn't necessary any more?

**Comment by fjetter** [comment]
OSX 3.9 not ci1 failure https://github.com/dask/distributed/issues/6076

File: distributed/deploy/tests/test_local.py
Comment (graingert) [reply distributed/deploy/tests/test_local.py:1074]: Nice catch 🤦

File: distributed/comm/tests/test_ucx.py
Comment (graingert) [reply distributed/comm/tests/test_ucx.py:299]: gen_test defaults to a 30 second timeout, this generally takes 24 seconds but I've seen it take 32

File: distributed/tests/test_worker.py
Comment (graingert) [reply distributed/tests/test_worker.py:1361]: cleanup adds `with clean:` which is already in `gen_test`
https://github.com/dask/distributed/blob/26ce9bbf4f2ab2faf632f7763bab4b8c81297acb/distributed/utils_test.py#L1767

File: distributed/tests/test_preload.py
Comment (fjetter) [inline distributed/tests/test_preload.py:170]: This feels a bit off. Why are we now writing a binary HTTP response to a plain socket?

File: distributed/tests/test_worker.py
Comment (fjetter) [reply distributed/tests/test_worker.py:1361]: the cleanup is part of gen_test. cleanup just calls a contextmanager that is also invoked by gen_test

File: distributed/tests/test_preload.py
Comment (graingert) [reply distributed/tests/test_preload.py:170]: It used to be spinning up a whole tornado server in a subprocess - which leaked a bunch of stuff

Comment (graingert) [reply distributed/tests/test_preload.py:170]: I can't use an async server because preload uses a blocking urllib3 request

File: distributed/deploy/tests/test_local.py
Comment (graingert) [reply distributed/deploy/tests/test_local.py:1067]: the code didn't need any blocking at all - it was unreachable because cluster raises the TypeError in the `__aenter__`

File: distributed/tests/test_preload.py
Comment (fjetter) [reply distributed/tests/test_preload.py:170]: I don't mind us spinning up a whole HTTP server. I consider this very hard to maintain and extremely difficult to extend. If the tornado server is leaking, we should think about how to replace it with something else but relying on plain sockets is not something I like to do for this even though this very particular application is easy enough to understand.
Let's say we'd like to protect the preload route with Auth. How would this test then look like? What if we want the preload route to accept additional parameters? There are a bunch of situations I can see happening that would require us to write much more complicated HTTP. If any other test dependencies can help us simplify this, that's something to consider.

Is this actually required to get pytest-asyncio out? I would rather let us ignore a few resource leakages.
If a blocking urllib3 request is the issue, maybe we should not have anything blocking in, in the first place and we should have a look at this, but at a later time.

File: distributed/deploy/tests/test_local.py
Comment (graingert) [reply distributed/deploy/tests/test_local.py:1067]: the code was added here, was it reachable when that PR was made and then later made unreachable? 

https://github.com/dask/distributed/pull/4733/files#diff-c9c4da87696cc1b04f54c8b199ebd277ca69da7eefcc08109d64349a4d2309a9R1081 

cc @fjetter

Comment (fjetter) [reply distributed/deploy/tests/test_local.py:1067]: The client in this test was never reachable. It was never intended to be reachable. I can't remember why I wrote the test in such a complicated fashion. I'm OK with the new version

File: distributed/tests/test_preload.py
Comment (graingert) [reply distributed/tests/test_preload.py]: @fjetter I had a look at vcrpy and decided it was too invasive - so I think a plain `mock.patch` on urllib3 is better

**Review by fjetter — APPROVED** [review]
LGTM once builds pass

**Comment by graingert** [comment]
test_client.py::test_reconnect timed out in py3.10 ci1

**Review by fjetter — APPROVED** [review]

**Pull request merged**
